src/Mvc/Mvc.Core/src/ModelBinding/Binders/CancellationTokenModelBinderProvider.cs

```diff
@@ -11,6 +11,8 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
     /// </summary>
     public class CancellationTokenModelBinderProvider : IModelBinderProvider
     {
+        private CancellationTokenModelBinder _modelBinder;
```

Comment:
```suggestion
        // CancellationTokenModelBinder  does not have any state. Re-use the same instance for binding.
        private readonly CancellationTokenModelBinder _modelBinder = new CancellationTokenModelBinder();
```

Reply:
Instantiating it up front seems fine. It's one additional instance per app, so I wouldn't be too worried about not allocating it. Same suggestion with the other type + the comment